rules/cross-platform/command_and_control_potential_gc2_sheet_usage.toml

```diff
@@ -0,0 +1,106 @@
+[metadata]
+creation_date = "2023/04/23"
+integration = ["endpoint"]
+maturity = "production"
+min_stack_comments = "New fields added: required_fields, related_integrations, setup"
+min_stack_version = "8.3.0"
+updated_date = "2023/04/23"
+
+[rule]
+author = ["Elastic"]
+description = """
+Detects the potential use of the Google Command and Control (GC2) tool. GC2 is a public open-source red teaming tool
+that uses Google Workspace services for C2. While the tool is only intended for red teaming, it has been leveraged by
+advanced persistent threats (APT) for intrusions.
+"""
+false_positives = ["    "]
+from = "now-9m"
+index = ["logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Potential Google Command and Control (GC2) Sheet Activity"
+references = ["https://github.com/looCiprian/GC2-sheet"]
+risk_score = 73
+rule_id = "d54a104e-e223-11ed-8d6e-f661ea17fbcd"
+severity = "high"
+tags = [
+    "Elastic",
+    "Host",
+    "Linux",
+    "Threat Detection",
+    "Command and Control",
+    "Exfiltration",
+    "Execution",
+    "Windows",
+    "Linux",
+    "macOS",
+]
+type = "eql"
+
+query = '''
+sequence by host.id with maxspan=1m
+
+    /* unsigned binary executed where process is spawned */
+    [process where host.os.type: ("windows", "macOS", "linux") and event.action == "start" and process.code_signature.exists == false]
+
+    /* 3rd party application using OAuth token */
+    [network where host.os.type: ("windows", "macOS", "linux") and event.action == "lookup_requested" and dns.question.name == "oauth2.googleapis.com"]
+
+    /* connection attempt to Google after OAuth authorization */
+    [network where host.os.type: ("windows", "macOS", "linux") and event.action == "connection_attempted" and destination.as.organization.name == "GOOGLE"]
+
+    /* DNS lookup for sheets.googleapis.com where sheet containing commands is located */
+    [network where host.os.type: ("windows", "macOS", "linux") and event.action == "lookup_requested" and
+        dns.question.name : ("sheets.googleapis.com", "drive.googleapis.com", "script.googleapis.com")]
+
+    [any where host.os.type: ("windows", "macOS", "linux") and
+
+        /* commands being executed via cmd or PS that were pulled from the Google Sheet for Windows */
+        /* file downloads from Drive with suspicious extensions for Windows */
+        (event.category == "process" and process.name: ("cmd.exe", "powershell.exe") and process.command_line: "cmd /c*") or
+        (event.category == "file" and event.action == "modification" and file.extension : ("exe","bat","cmd","vbs","js","ps1","hta","scr","dll"))]
```

Comment:
Does this sequence trigger in Linux and MacOS? Maybe the windows specific stuff, like the last event in the sequence, will prevent it from doing so

Reply:
Hey @w0rk3r! - Yea I still have some additional research to do but wanted to capture the logic momentarily. I plan to build the binary on macOS and Linux sometime this week and update where applicable. I'd like to also stay clear of `dns.question.name` but that seems to be the only network traffic indicator from the data itself.